pywbem/_logging.py

```diff
@@ -117,279 +167,183 @@
 
-            * ``detail_level``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DETAIL_LEVELS` list.
+    Examples::
 
-            * ``dest``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DESTINATIONS` list.
+        ops=stderr    # set cim operations logger
 
-          log_filename (:term:`string`)
-            Optional string that defines the filename for output of logs
-            if the dest type is `file`
+        http=file     # set http logger to send to file
 
-        Raises:
+        all=file      # Set all loggers to output to a file
+    """  # noqa: E501
 
-          ValueError: Generated if the syntax of the input string is invalid
-            or any of the components is not one of allowed strings.
+    log_specs = log_spec_str.split(',')
+    for log_spec in log_specs:
+        try:
+            spec_split = log_spec.split("=", 1)
+        except ValueError:
+            raise ValueError('Log spec %s invalid. Contains too many '
+                             ' components' % log_spec)
+        if len(spec_split) == 1:
+            log_component = spec_split[0]
+            log_dest = ''
+        elif len(spec_split) == 2:
+            log_component = spec_split[0]
+            log_dest = spec_split[1]
+        else:
+            raise ValueError("Log component name required in %s" % log_spec)
 
-        Examples::
+        # cvt empty strings to None
+        log_dest = 'none' if log_dest == '' else log_dest
+        define_logger(log_component, log_dest=log_dest,
+                      log_filename=log_filename)
 
-            ops=stderr:min    # set cim operations logger
 
-            http=file:        # set http logger to send to file
+def define_logger(log_component, log_dest=DEFAULT_LOG_DESTINATION,
+                  log_filename=DEFAULT_LOG_DESTINATION):
+    """
+    This is a helper function to define a single pywbem logger with the
+    logger name defined by log_component
+    Create create the logger with the name defined by log_component and with
+    the parameters defined by log_dest and log-filename.
+
+    This function can be used to set up all of the named loggers used by
+    pywbem.
+
+    This function sets up the logger name, and a formatter and also defines
+    the log destination.
+
+    Parameters:
+
+      log_component (:term:`string`):
+        The name of the logger. It must be one of the
+        names defined in :data:`~pywbem._logging.LOG_COMPONENTS`.
+        Used to create the logger name by prepending with the logger name
+        prefix ``pywbem.``.  The log_component may include a suffix component
+        to uniquely identify a logger. This is identified with  a period
+        between the log_component name and the suffix. Normally this is
+        used to define a logger specifically for a selected connection and
+        the suffix is the conn_id of the WBEMConnection.
```

Comment:
On "Normally...": There is no choice the user has. The code uses the connection id, so the user has to either omit it, or add exactly the connection id.

Reply:
Done